app/src/main/java/com/example/ithaca_transit_android_v2/MainActivity.kt

```diff
@@ -1,50 +1,134 @@
 package com.example.ithaca_transit_android_v2
 
+import android.content.Intent
 import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
 import android.util.Log
+import android.view.View
+import android.widget.AdapterView
+import android.widget.EditText
 import androidx.appcompat.app.AppCompatActivity
-import com.example.ithaca_transit_android_v2.models.Coordinate
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.async
-import kotlinx.coroutines.runBlocking
+import com.example.ithaca_transit_android_v2.models.Location
+import com.example.ithaca_transit_android_v2.states.*
+import com.example.ithaca_transit_android_v2.ui_adapters.SearchViewAdapter
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import io.reactivex.Observable
+import io.reactivex.ObservableEmitter
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.activity_toolbar_search.*
+import kotlinx.android.synthetic.main.item_searchview.*
 
+class MainActivity : AppCompatActivity(), OnMapReadyCallback {
+    private lateinit var disposable: Disposable
+    private var mSearchLocations: List<Location> = ArrayList()
+    private var mSearchAdapter: SearchViewAdapter? = null
 
-class MainActivity : AppCompatActivity() {
+    override fun onMapReady(map: GoogleMap?) {
+        map!!.setOnMapClickListener { point ->
+            Log.i("qwerty","map clicked")
+            search_view.clearFocus() }
+    }
+
+    private fun createSearchObservable(): Observable<SearchState> {
```

Comment:
i feel like these functions shouldn't be implemented in the `MainActivity` and instead implemented in their respective presenter classes...

Reply:
Made an issue